collated/main/A0138411N.md

```diff
@@ -0,0 +1,877 @@
+    }
+
+    private void updateFilteredListToShowDate(Expression expression) {
+        filteredTasks.setPredicate(expression::satisfies);
+    }
+```
+###### /main/java/seedu/ggist/model/ModelManager.java
+``` java
+    private class DateQualifier implements Qualifier {
+        private String taskDateKeyWords;
+
+        DateQualifier(String taskDateKeyWords) {
+            this.taskDateKeyWords = taskDateKeyWords;
+        }
+
+        @Override
+        public boolean run(ReadOnlyTask task) {
+            if (taskDateKeyWords == null) {
+                return true;
+            }
+            return ((taskDateKeyWords.equalsIgnoreCase(task.getStartDate().toString()) || 
+                   taskDateKeyWords.equalsIgnoreCase(task.getEndDate().toString())) && !task.isDone()) ||
+                   ((task.getStartDate().value.equals(Messages.MESSAGE_NO_START_DATE_SPECIFIED) && 
+                   (task.getStartTime().value.equals(Messages.MESSAGE_NO_START_TIME_SET) &&
+                   (task.getEndDate().value.equals(Messages.MESSAGE_NO_END_DATE_SPECIFIED) && 
+                   (task.getEndTime().value.equals(Messages.MESSAGE_NO_END_TIME_SET)) && !task.isDone())) ||
+                   (task.isOverdue() && !task.isDone())));
+        }
+```
+###### /main/java/seedu/ggist/model/task/DeadlineTask.java
+``` java
+    /**
+     * Every field must be present and not null.
+     * @throws IllegalValueException 
+     */
+    public DeadlineTask(TaskName taskName, TaskDate taskDate, TaskTime startTime, TaskDate endDate, TaskTime endTime, Priority priority) throws IllegalValueException {
+        super(taskName, taskDate, startTime, endDate, endTime, priority);
+    }
+    
+    /**
+     * Copy constructor.
+     * @throws IllegalValueException 
+     */
+    public DeadlineTask(ReadOnlyTask source) throws IllegalValueException {
+        this(source.getTaskName(), source.getStartDate(), source.getStartTime(), source.getEndDate(), source.getEndTime(), source.getPriority());
+    }
+    
+    @Override
+    public int hashCode() {
+        // use this method for custom fields hashing instead of implementing your own
+        return Objects.hash(taskName, startDate, startTime, endDate, endTime, priority);
+    }
+}
+```
+###### /main/java/seedu/ggist/model/task/EventTask.java
+``` java
+    /**
+     * Every field must be present and not null.
+     * @throws IllegalValueException 
+     */
+    public EventTask(TaskName taskName, TaskDate startDate, TaskTime startTime, TaskDate endDate, TaskTime endTime, Priority priority) throws IllegalValueException {
+        super(taskName, startDate, startTime, endDate, endTime, priority);
+    }
+    
+    public EventTask(TaskName taskName, TaskDate startDate, TaskTime startTime, TaskDate endDate, TaskTime endTime, Priority priority, boolean done, boolean overdue) throws IllegalValueException {
+        super(taskName, startDate, startTime, endDate, endTime, priority);
+        this.done = done;
```

Comment:
need to use better names to the boolean variables indicative of their boolean nature. please take a look at the coding guidelines